docs/docs/local-https.md

```diff
@@ -8,50 +8,148 @@ Gatsby provides an easy way to use a local HTTPS server during development, than
+
 ## Custom Key and Certificate Files
 
 You may find that you need a custom key and certificate file for https if you use multiple
 machines for development (or if your dev environment is containerized in Docker).
 
-If you need to use a custom https setup, you can pass the `--https`, `--key-file` and
-`--cert-file` flags to `npm run develop`.
+If you need to use a custom https setup, you can pass the `--https`, `--key-file`,
+`--cert-file`, and `--ca-file` flags to `npm run develop`.
 
-- `--cert-file` [relative path to ssl certificate file]
-- `--key-file` [relative path to ssl key file]
+- `--cert-file` [relative/absolute path to ssl certificate file]
+- `--key-file` [relative/absolute path to ssl key file]
+- `--ca-file` [relative/absolute path to ssl certificate authority file]
 
-See the example command:
+See the example commands below:
 
-```shell
-gatsby develop --https --key-file ../relative/path/to/key.key --cert-file ../relative/path/to/cert.crt
-```
+- Using `npm run develop`
+
+  ```shell
+  # Using relative paths
+  $ npm run develop -- --https --key-file ../relative/path/to/key.key --cert-file ../relative/path/to/cert.crt --ca-file ../relative/path/to/ca.crt
+
+  # Or using absolute paths
+  $ npm run develop -- --https --key-file /absolute/path/to/key.key --cert-file /absolute/path/to/cert.crt --ca-file /absolute/path/to/ca.crt
+  ```
+
+- Alternatively, you can run the development server using the gatsby cli
+
+  ```shell
+  # Using relative paths
+  $ gatsby develop --https --key-file ../relative/path/to/key.key --cert-file ../relative/path/to/cert.crt --ca-file ../relative/path/to/ca.crt
+
+  # Or using absolute paths
+  $ gatsby develop --https --key-file /absolute/path/to/key.key --cert-file /absolute/path/to/cert.crt --ca-file /absolute/path/to/ca.crt
+  ```
+
+Usage of the `--ca-file` flag is only required if your certificate is signed by a certificate authority.
+
+If your certificate is self-signed, then do not include the `--ca-file` flag. Also, if you want your browser to trust a self-signed certificate, you will need to add it to your operating system (or browser's, in Firefox's case) root certificate store in order for your browser to trust it.
 
-in most cases, the `--https` passed by itself is easier and more convenient to get local https.
+In most cases, the `--https` passed by itself is easier and more convenient to get local https.
 
 ---
 
-Keep in mind that the automatic certificates issued with the `--https` flag are explicitly issued to `localhost` and will only be accepted there. Using it together with the `--host` option will likely result in browser warnings.
+Automatic certificates issued with the `--https` flag are issued to `localhost` by default, unless you have used the `--host` flag. If you have, a record in your hosts file will automatically be configured to point the defined host to `127.0.0.1`. At this time, ip addresses defined by `--host` are not supported.
```

Comment:
```suggestion
Automatic certificates issued with the `--https` flag are issued to `localhost` by default, unless you have used the `--host` flag. If you have, a record in your hosts file, it will automatically be configured to point the defined host to `127.0.0.1`. At this time, ip addresses defined by `--host` are not supported.
```

Reply:
Should "a record in your hosts file, " be removed?

Reply:
About removing that phrase: this process is not contingent on there being a record in the hosts file, so saying "if you have a record in your hosts file, it will automatically be configured" wouldn't be accurate.

If we remove "a record in your hosts file, ", then "it" becomes kind of vague in this context... what is "it"?  If the goal is to make it a bit less technical, maybe it could read?
```diff
- If you have, it will automatically be configured
+ If you have, your device will automatically be configured
```

Reply:
@wardpeet any thoughts?